pyproject.toml

```diff
@@ -12,29 +12,32 @@ classifiers = [
     'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
     'Programming Language :: Python :: 3.11',
+    'Programming Language :: Python :: 3.12',
     'Topic :: Scientific/Engineering :: Artificial Intelligence',
 ]
 keywords = ['sdv', 'synthetic-data', 'synhtetic-data-generation', 'timeseries', 'single-table', 'multi-table']
 dynamic = ['version']
 license = { text = 'BSL-1.1' }
-requires-python = '>=3.8,<3.12'
+requires-python = '>=3.8,<3.13'
 readme = 'README.md'
 dependencies = [
-    'boto3>=1.15.0',
-    'botocore>=1.18',
+    'boto3>=1.28',
+    'botocore>=1.31',
     'cloudpickle>=2.1.0',
     'graphviz>=0.13.2',
     "numpy>=1.20.0;python_version<'3.10'",
-    "numpy>=1.23.3;python_version>='3.10'",
+    "numpy>=1.23.3,<2;python_version>='3.10' and python_version<'3.12'",
+    "numpy>=1.26.0,<2;python_version>='3.12'",
     "pandas>=1.1.3;python_version<'3.10'",
     "pandas>=1.3.4;python_version>='3.10' and python_version<'3.11'",
-    "pandas>=1.5.0;python_version>='3.11'",
-    'tqdm>=4.15',
-    'copulas>=0.9.0',
-    'ctgan>=0.9.0',
-    'deepecho>=0.5',
+    "pandas>=1.5.0;python_version>='3.11' and python_version<'3.12'",
+    "pandas>=2.0.2;python_version>='3.12'",
```

Comment:
This is required due to [this](https://github.com/sdv-dev/SDV/issues/1469#issuecomment-1745675813) issue.